src/structs/Api.ts

```diff
@@ -39,9 +40,10 @@ export class Api extends EventEmitter {
   constructor(token: string, options: APIOptions = {}) {
     super();
     this.options = {
-      token: token,
+      token,
       ...options,
     };
+    this.id = JSON.parse(atob(token.split(".")[1])).id;
```

Comment:
would recommend making a function that validates the JWT before fetching the payload.

```js
function destructToken(token: string) {
  const segments = token.split('.');
  if (segments.length !== 3) {
    throw new Error("malformed token");
  }
  
  const payload = atob(token[1]);
  try {
    const json = JSON.parse(payload);
    return json.id;
  } catch(err) {
    throw new Error("invalid json in token", err);
  }
  
  throw new Error("invalid state, this should be impossible! Please report!");
}